build-infra/build-vm.sls

```diff
@@ -144,3 +165,109 @@ commands-keyring:
     - runas: user
     - onchange:
       - file: /home/user/trusted-keys-for-commands.asc
+
+/home/user/.ssh/config:
+  file.managed:
+    - user: user
+    - contents: |
+{% for host, config in salt['pillar.get']('build-infra:remote-hosts', {}).items() %}
+        Host {{host}}
+          HostName {{host}}
+          User {{config.ssh_user}}
+{% endfor %}
+    - mode: 0755
+    - makedirs: True
+
+{% for host, config in salt['pillar.get']('build-infra:remote-hosts', {}).items() %}
+{{host}}:
+  ssh_known_hosts.present:
+    - user: user
+    - enc: ssh-rsa
+    - key: {{config.ssh_host_key}}
+    - hash_known_hosts: False
+{% endfor %}
+
+github.com:
+  ssh_known_hosts.present:
+    - user: user
+    - enc: ssh-rsa
+    - key: AAAAB3NzaC1yc2EAAAABIwAAAQEAq2A7hRGmdnm9tUDbO9IDSwBK6TbQa+PXYPCPy6rbTrTtw7PHkccKrpp0yVhp5HdEIcKr6pLlVDBfOLX9QUsyCOV0wzfjIJNlGEYsdlLJizHhbn2mUjvSAHQqZETYP81eFzLQNnPHt4EVVUh7VfDESU84KezmD5QlWpXLmvU31/yMf+Se8xhHTvKSCZIFImWwoG6mbUoWf9nzpIoaSjB+weqqUUmpaaasXVal72J+UX2B+2RPW3RcT0eOzQgqlJL3RKrTJvdsjE3JEAvGq3lGHSZXy28G3skua2SmVi/w4yCE6gbODqnTWlg7+wC604ydGXA8VJiS5ap43JXiUFFAaQ==
+    - hash_known_hosts: False
+
+{% if salt['pillar.get']('build-infra:mirror_ssh_key') %}
+/home/user/.ssh/id_rsa:
+  file.managed:
+    - contents_pillar: build-infra:mirror_ssh_key
+    - mode: 600
+    - user: user
+    - group: user
+    - makedirs: True
+    - dir_mode: 700
+{% endif %}
+
+{% for builder in builders_list %}
+{% set config_baseurl = salt['pillar.get']('build-infra:build-envs:' + env + ':builders-list:' + builder + ':config:repository:baseurl', 'https://github.com/QubesOS/qubes-') %}
+{% set config_repo = salt['pillar.get']('build-infra:build-envs:' + env + ':builders-list:' + builder + ':config:repository:component', 'release-configs') %}
+{% set config_file = salt['pillar.get']('build-infra:build-envs:' + env + ':builders-list:' + builder + ':config:file') %}
+{% set keys =  salt['pillar.get']('build-infra:build-envs:' + env + ':builders-list:' + builder + ':keys', []) %}
+
+{{builder}}-get:
+  git.latest:
+    - name : https://github.com/QubesOS/qubes-builder
+    - target: {{ builder }}
+    - user: user
+
+{{builder}}-check:
+  cmd.run:
+    - name: git verify-tag --raw "$(git describe --abbrev=0)" 2>&1 >/dev/null | grep '^\[GNUPG:\] TRUST_FULLY'
```

Comment:
`--abbrev=0` here is wrong. It will verify latest available tag, not the tag describing that is downloaded. In other words - it won't fail if some malicious commits are added at the top.

Reply:
Should it be: ```git verify-commit --raw HEAD```?

Reply:
verify-commit and verify-tag are two different things. It should be tag here (otherwise contributor's commit on top, only signed with my tag wouldn't be accepted).